### PR TITLE
React hooks axios post

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description

Lesson on Axios.post with hooks.

## Questions for the Author
- [ ] Have you updated the weekly learning goals in the pedagogy resource?
- [ ] Have you updated the weekly assessment in Schoology?
- [x] Have you added links to the source (lucidchart, draw.io) for any diagram?
- [x] Have you updated the READMEs with links to new or moved resources?

## Feedback Requested
Please utilize the built-in GitHub Reviewers functionality to assign necessary reviewers. Use this section to further specify portions of the PR that you would like additional or focused feedback on.

### CODE REVIEW

File: React/apis-get.md [hunk 1]
```diff
@@ -3,16 +3,16 @@
 ## Learning Goals
 
 - Introduce `axios` as our API library
-- Integrate a get API request in react lifecycle method
+- Integrate a get API request in react lifecycle method `useEffect`
```
Comment (kaidamasaki) [inline React/apis-get.md:6]:
```suggestion
- Integrate a get API request into the react lifecycle using the `useEffect` method
```

File: React/apis-get.md [hunk 2]
```diff
@@ -25,50 +25,52 @@ We have the json-server package pre-installed and set up in our `package.json` f
...
-    "server": "json-server --watch src/data/pets.json",
+    "server": "json-server --watch src/data/students.json",
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },
 ```
 
-Check out the `server` line.  This adds a command to start the json-server package on the port 3000.  The command is: `npm run server`.  You can open the api in the browser with [http://localhost:3000/pets](http://localhost:3000/pets).  You can do all the RESTful routes for a pets resource, listing showing, creating etc.  All these operations will update the pets.json file.  So `pets.json` will serve as an imitation API for us to learn with.
+Check out the `server` line.  This adds a command to start the json-server package on the port 3000.  The command is: `npm run server`.  You can open the api in the browser with [http://localhost:3000/students](http://localhost:3000/students).  You can do all the RESTful routes for a students resource, listing showing, creating etc.  All these operations will update the students.json file.  So `students.json` will serve as an imitation API for us to learn with.
```
Comment (kaidamasaki) [inline React/apis-get.md:35]: Maybe add a comment about how this should be familiar from ada-pets-node?

File: React/apis-post.md [hunk 3]
```diff
@@ -6,50 +6,63 @@
...
-    const maxId = Math.max(...petIds);
-    pet.id = maxId + 1;
-    petList.push(pet);
+    const newStudentList = [...students];
 
-    this.setState(petList);
+    // Find the max id and add 1
+    const nextId = newStudentList.reduce((accumulator, currentStudent) => {
+      return Math.max(accumulator, currentStudent.id);
+    }, 0) + 1;
```
Comment (kaidamasaki) [inline React/apis-post.md:24]:
```suggestion
    const studentIds = newStudentList.map(student => student.id);
    const nextId = Math.max(...studentIds) + 1;
```

File: React/apis-post.md [hunk 4]
```diff
@@ -60,16 +72,22 @@ Now observe what happens when we submit the form in our application. Check in wi
 - How can you handle validation errors in a real API, like we made before?
 - How could you delete pets from the API with your React App?
   - With your neighbor implement this functionality.
-  - Verify that the pets dissapear from the json file.
 
-When you are ready you can see [our solution](examples/deletePet.js).
+## Exercise
+
+At this point you can list students and add new students, but you cannot delete them.  For this exercise take the working code from this [point in the textbook](https://github.com/AdaGold/ada-students/tree/12-api-post), and add a `delete` button to each student and give the app the ability to delete students.  This should cause a `DELETE` request to be made from the API.
```
Comment (tildeee) [inline React/apis-post.md:78]:
```suggestion
At this point you can list students and add new students, but you cannot delete them.  For this exercise take the working code from this [point in the textbook](https://github.com/AdaGold/ada-students/tree/12-api-post), and add a `delete` button to each student and give the app the ability to delete students.  Every time a user clicks on a "delete" button for a specific student, our app should send a `DELETE` request to the API.  
```

The phrase "This should cause a `DELETE` request to be made from the API" should be changed to "This should send a `DELETE` request to the API". Optionally, I've suggested changing the first part "This" to "every time a user clicks on a delete button"

Comment (kaidamasaki) [reply React/apis-post.md:78]: They will have used a `DELETE` request in [Wave 3 of Ada Pets Node](https://github.com/AdaGold/ada-pets-node#wave-3-remove-pet).  

Though we maybe should have added more info there.

**Pull request merged**